evalai/utils/challenges.py

```diff
@@ -84,3 +84,81 @@ def get_future_challenge_list():
     """
     url = "{}{}".format(API_HOST_URL, Urls.future_challenge_list.value)
     get_challenges(url)
+
+
+def get_teams(url):
+    """
+    Returns the teams corresponding to the user.
+    """
+    headers = get_headers()
+
+    try:
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+    except requests.exceptions.HTTPError as err:
+        echo(err)
+        sys.exit(1)
+    except requests.exceptions.RequestException as err:
+        echo(err)
+        sys.exit(1)
+
+    response_json = response.json()
+
+    if valid_token(response_json):
+        return response_json['results']
+    else:
+        echo("The token is not valid. Try again.")
+        sys.exit(1)
+
+
+def get_teams_challenges(url, teams):
+    """
+    Returns the challenges corresponding to the teams.
+    """
+    challenges = []
+    for team in teams:
+        headers = get_headers()
+        try:
+            response = requests.get(url.format(team['id']), headers=headers)
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as err:
+            echo(err)
+            sys.exit(1)
+        except requests.exceptions.RequestException as err:
+            echo(err)
+            sys.exit(1)
+        response_json = response.json()
+        challenges = challenges + response_json['results']
+    return challenges
+
+
+def get_challenge_count(mode):
+    """
+    Gets the challenge the user has participated or hosted.
+    """
+    challenges = []
+
+    if mode == 'host':
+        url = Urls.host_teams.value
+        url = "{}{}".format(API_HOST_URL, url)
+        teams = get_teams(url)
+        if not teams:
+            echo("The token is not valid. Try again.")
+
+        url = "{}{}".format(API_HOST_URL, Urls.host_challenges.value)
+        challenges = get_teams_challenges(url, teams)
+    elif mode == 'participate':
+        url = Urls.participant_teams.value
+        url = "{}{}".format(API_HOST_URL, url)
+        teams = get_teams(url)
+        if not teams:
+            echo("The token is not valid. Try again.")
+
+        url = "{}{}".format(API_HOST_URL, Urls.participant_challenges.value)
+        challenges = get_teams_challenges(url, teams)
+    else:
+        echo("Option doesn't exist. Use --help for information")
+
+    for challenge in challenges:
+        challenge = print_challenge_table(challenge)
+        echo(challenge)
```

Comment:
What does this echo do? Aren't you printing the challenge in `print_challenge_table`?
Can you see the fallacy here? The function `print_x` does not print and is instead returning something to be printed. The right thing to do would be to just print it inside the `print_x` function and not return anything.

Reply:
That was my mistake, I wanted it to return output so that I could use it for testing and then I used a different method and I forgot about this.